src/main/java/org/snomed/snowstorm/rest/ConceptController.java

```diff
@@ -387,17 +391,24 @@ public ConceptReferencesResult findConceptReferences(
 	public ResponseEntity<ConceptView> createConcept(
 			@PathVariable String branch,
 			@RequestParam(required = false, defaultValue = "false") boolean validate,
-			@RequestBody @Valid ConceptView concept,
-			@RequestHeader(value = "Accept-Language", defaultValue = Config.DEFAULT_ACCEPT_LANG_HEADER) String acceptLanguageHeader) throws ServiceException, JsonProcessingException {
+			@RequestBody ConceptView concept,
+			@RequestHeader(value = "Accept-Language", defaultValue = Config.DEFAULT_ACCEPT_LANG_HEADER) String acceptLanguageHeader) throws ServiceException {
 
 		if (validate) {
-			final List<InvalidContent> invalidContentWarnings = ConceptValidationHelper.validate((Concept) concept, BranchPathUriUtil.decodePath(branch), validationService, objectMapper);
-			final Concept createdConcept = conceptService.create((Concept) concept, ControllerHelper.parseAcceptLanguageHeaderWithDefaultFallback(acceptLanguageHeader),
-					BranchPathUriUtil.decodePath(branch));
-			return new ResponseEntity<>(createdConcept, ConceptValidationHelper.getCreateConceptHeaders(ControllerHelper.getCreatedLocationHeaders(createdConcept.getId()),
-					objectMapper.writeValueAsString(ConceptValidationHelper.replaceTemporaryUUIDWithSCTID(invalidContentWarnings, createdConcept))), HttpStatus.OK);
+			final InvalidContentWithSeverityStatus invalidContent = ConceptValidationHelper.validate((Concept) concept, BranchPathUriUtil.decodePath(branch), validationService);
+			if (invalidContent.getSeverity() == Severity.WARNING) {
+				final Concept createdConcept = conceptService.create((Concept) concept, ControllerHelper.parseAcceptLanguageHeaderWithDefaultFallback(acceptLanguageHeader),
+						BranchPathUriUtil.decodePath(branch));
+				// validate concept after creation due to temporary id provided by UI can exceed the concept id range.
+				validator.validate(createdConcept);
```

Comment:
Is this just the simple bean validation, not the expensive drools call? It's not clear when reading the code, maybe the naming could be adjusted somehow?

Reply:
Good suggestion, fixed.